tree/ntuple/v7/src/RField.cxx

```diff
@@ -223,7 +223,33 @@ void ROOT::Experimental::Detail::RFieldBase::TraverseVisitor(RNTupleVisitor &vis
 
 void ROOT::Experimental::Detail::RFieldBase::AcceptVisitor(Detail::RNTupleVisitor &visitor, int level) const
 {
-    visitor.VisitField(*this, level);
+   visitor.VisitField(*this, level);
+}
+
+void ROOT::Experimental::Detail::RFieldBase::TraverseValueVisitor(RValueVisitor &visitor, int level) const
+{
+   // subfields of a std::vector and std::array shouldn't be displayed
+   if ((GetLevelInfo().GetLevel() != 0) && (GetParent()->GetStructure() == ENTupleStructure::kCollection || GetParent()->GetType().compare(0, 11, "std::array<") == 0 )) return;
```

Comment:
Better check for `fNRepetitions > 0` instead of `std::array` because we'll have other variations of fixed-sized arrays in the future.